File: BehaviorPattern/Chain_Of_Responsibility/Golang/Concretes/Level2SupportHandler.go
package concretes

import (
    "fmt"
    "v"
    "v/Interfaces"
)

type Level2SupportHandler struct {
    nextHandler interfaces.SupportHandler
}

func (h *Level2SupportHandler) SetNext(handler interfaces.SupportHandler) {
    h.nextHandler = handler
}

func (h *Level2SupportHandler) HandleRequest(request *v.SupportRequest) {
    if request.GetComplexity() <= 2 {
        fmt.Println("Level 2 support handled the request:", request.GetMessage())
    } else if h.nextHandler != nil {
        h.nextHandler.HandleRequest(request)
    }
}

File: BehaviorPattern/Chain_Of_Responsibility/Python/Concretes/Level2SupportHandler.py
from Interfaces.SupportHandler import SupportHandler
from .SupportRequest import SupportRequest

class Level2SupportHandler(SupportHandler):
    def handle_request(self, request: SupportRequest):
        if request.get_complexity() <= 2:
            print(f"Level 2 support handled the request: {request.get_message()}")
        elif self.next_handler is not None:
            self.next_handler.handle_request(request)
